coderadar-server/coderadar-core/src/main/java/org/wickedsource/coderadar/metricquery/rest/tree/MetricsTreeQuery.java

```diff
@@ -5,20 +5,23 @@
 import java.util.List;
 import javax.validation.constraints.NotNull;
 import javax.validation.constraints.Size;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
 
 /**
  * Provides parameters to query for values of one or more metrics values at the time of a specific
  * commit.
  */
+@NoArgsConstructor
 public class MetricsTreeQuery {
 
-  @NotNull private String commit;
+  @Getter @Setter @NotNull private String commit;
```

Comment:
is the setter necessary?

Reply:
The setter is only used in one test, but it will fail without it, as it's required for the RequestBody to have meaningful data.